packages/components/input/src/input.js

```diff
@@ -8,16 +9,38 @@ export class TSInput extends TSElement {
 
 	static get properties() {
 		return {
-			/** Disable status of input */
-			disabled: { type: Boolean, attribute: 'disabled', reflect: true }
+			/** Show error style */
+			hasError: { type: Boolean, reflect: true },
+			/** Disable state of the input */
+			disabled: { type: Boolean, reflect: true },
+			/** Readonly state of the input */
+			readonly: { type: Boolean, reflect: true },
+			/** Direction 'rtl' or 'ltr' */
+			dir: { type: String, reflect: true },
+			/** Icon that appears at the end part of the input. Readonly and disabled state will show a lock icon instead. */
+			iconRight: { type: String, reflect: true, attribute: 'icon-right' },
+			/** Icon that appears at the beginning of the input */
+			iconLeft: { type: String, reflect: true, attribute: 'icon-left' }
```

Comment:
I suggest to change naming to "iconStart" and "iconEnd" so it would be more clear in case of RTL.